### PR TITLE
Add Aftermath mission In the Nick of Time

### PR DESCRIPTION
**Pull request by Smittytron**
Einstein needs to stop getting captured.

### CODE REVIEW

**Comment by darcyrush** [comment]
Mission plays well.

Minor suggestion; perhaps scientist and Einstein selection priority could be lowered

**Comment by Smittytron** [comment]
> perhaps scientist and Einstein selection priority could be lowered

IMO that'd create an equally frustrating issue where players intend to grab the scientists and go but one or more combat units were in their dragbox.

**Comment by tttppp** [comment]
Mission was good. At first it seemed excessively easy (jumping chronotanks into an empty base), but the scientist half of the mission was a bit harder.

On my first attempt I lost by destroying the forward command that the scientists come out of. I didn't spot anything in the briefing telling me not to blow this up and was confused as to why I lost. Maybe the failure message could be a bit more explicit: "Forward command containing scientists destroyed - mission failed."

The spy seems a bit pointless. I managed to disguise my spy and then immediately ran him onto a mine!

After the scientists are released then a Allied naval transport drops off some tanks, and these block one of the Soviet naval transports, sinking it before it can unload. I don't know if this is deliberate, but I would have expected to perform some quick thinking/micro to achieve this sort of thing rather than simply sitting and watching.

It's a bit weird losing control of my tanks at the extraction point. Is this deliberate?  I know the scientists are meant to leave the map like this, but I wanted to keep my tanks just in case I needed them.

File: mods/ra/maps/in-the-nick-of-time/in-the-nick-of-time.lua [hunk 1]
```diff
@@ -0,0 +1,412 @@
...
+StartUnits = { "jeep", "jeep", "1tnk", "1tnk", "1tnk", "1tnk", "2tnk", "2tnk" }
+StartingPlanes = { Yak1, Yak2 }
+ReinforcementFootprint1 = { CPos.New(93,78), CPos.New(94,78), CPos.New(95,78) }
+InfantrySquad = { "e1", "e1", "e1", "e3", "e3" }
+ReinforcementFootprint2 = { CPos.New(63,84), CPos.New(63,85), CPos.New(63,86), CPos.New(63,87) }
+SouthWaterPath1 = { SouthWaterEntry.Location, SouthWaterLanding1.Location }
+SouthWaterPath2 = { SouthWaterEntry.Location, SouthWaterLanding2.Location }
+SouthWaterTeam = { "dtrk", "1tnk", "1tnk", "2tnk", "2tnk" }
+SouthChinookPath = { ChinookEntry1.Location, ChinookLZ1.Location }
+SouthChinookChalk = { "e3", "e3", "e3", "e1", "e1" }
```
Comment (tttppp) [inline mods/ra/maps/in-the-nick-of-time/in-the-nick-of-time.lua:18]: This took me a few attempts to get the reinforcements and was super disappointing.  I think it's better to just let the nuke truck blow it up.

**Comment by tttppp** [comment]
Lua code looks good to me.

**Comment by Smittytron** [comment]
> On my first attempt I lost by destroying the forward command that the scientists come out of. I didn't spot anything in the briefing telling me not to blow this up and was confused as to why I lost. Maybe the failure message could be a bit more explicit: "Forward command containing scientists destroyed - mission failed."
> 
> After the scientists are released then a Allied naval transport drops off some tanks, and these block one of the Soviet naval transports, sinking it before it can unload. I don't know if this is deliberate, but I would have expected to perform some quick thinking/micro to achieve this sort of thing rather than simply sitting and watching.
> 
> It's a bit weird losing control of my tanks at the extraction point. Is this deliberate? I know the scientists are meant to leave the map like this, but I wanted to keep my tanks just in case I needed them.

Updated. I added a fail message explaining the scientists were in the command center. Changed the beach so the transport wouldn't be blocked. (Unless the player knows where the LZs are and intentionally blocks them. We can't fix that.) And I switched the ending to have only the scientists leave the map.

File: mods/ra/maps/in-the-nick-of-time/in-the-nick-of-time.lua (hunk 1 above)
Comment (Smittytron) [reply mods/ra/maps/in-the-nick-of-time/in-the-nick-of-time.lua:18]: > The spy seems a bit pointless. I managed to disguise my spy and then immediately ran him onto a mine!

Every reinforcement is true to the original.

**Review by tttppp — DISMISSED** [review]
I retested this, and everything is fixed. LGTM

**Comment by Smittytron** [comment]
Rebased

**Review by tttppp — APPROVED** [review]
Retested and it still looks good!  Does this need retesting every rebase??

**Comment by Smittytron** [comment]
> Retested and it still looks good! Does this need retesting every rebase??

Nah, a rebase with missions will always be the missions.yaml file. I just report 'Rebased' so that people know I didn't change anything related to the mission.

**Review by reaperrr — APPROVED** [review]

**Comment by reaperrr** [comment]
https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/f6ea7c76d272d5ecbbae1f7d9d76cdce47a5fbe0...29160b3df3171ba7b55b9687ddb986581c885371

**Pull request merged**